extensions/ql-vscode/src/cli.ts

```diff
@@ -482,6 +482,36 @@ export class CodeQLCliServer implements Disposable {
     return await this.runJsonCodeQlCliCommand<QuerySetup>(['resolve', 'library-path'], subcommandArgs, 'Resolving library paths');
   }
 
+  /**
+   * Finds the language that a query is analyzing.
+   * @param queryUri The URI of the query
+   */
+  async findLanguage(workspaces: string[], queryUri?: Uri): Promise<string> {
+    let language = '';
+    if (queryUri) {
+      const subcommandArgs = [
+        '--format', 'bylanguage',
+        queryUri.fsPath,
+        '--additional-packs',
+        workspaces.join(path.delimiter)
+      ];
+      try {
+        const JSONResults = await this.runCodeQlCliCommand(['resolve', 'queries'], subcommandArgs, 'Finding query language');
+
+        // This corresponds to `codeql resolve queries <query> --format bylanguage | jq -r '.byLanguage | keys[]'`
+        language = Object.keys(JSON.parse(JSONResults).byLanguage)[0];
```

Comment:
I would slightly lean towards adding a new type like we have for other CLI commands (e.g. `TestCompleted`) to capture the output of `codeql resolve queries --format bylanguage`) and then using that result type rather than doing this.

Reply:
To elaborate a bit on why I think that's better: Capturing the whole result of the comment rather than just doing some ad-hoc parsing of it here means that if in future we want to call that command for some other reason that uses another part of the output, it will be much easier to do.

Reply:
That makes sense, thanks! 🎉 I'll have a go at capturing the result as a type 😃